### PR TITLE
Adds the Uptane auto provisioner feature

### PR DESCRIPTION
**Pull request by heartsucker**
Creates Uptane director structure and pulls down the `root.json` from the API.
Does not validate the JSON is even JSON, let alone whether or not it is even a valid Uptane `root.json`.

This is a WIP that is going to need two other branches merged first and there are likely to be conflicts.

### CODE REVIEW

File: src/bin/sota_provisioner.rs [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    /// version.
+    fn check_root_json(&self) -> Result<(), ()> {
+        let has_root = self.check_fs_for_root_json()?;
+        if !has_root {
+            let root_meta_bytes = self.get_root_json()?;
+            let json_path = format!("{}/repo/director/metadata/current/root.json", self.config.uptane.repo_path);
+            let mut file = File::create(json_path).map_err(|_| ())?;
+            let perms = Permissions::from_mode(0o0640);
+            // Set safe perms before writing
+            let _ = file.set_permissions(perms).map_err(|_| ())?;
```
Comment (taheris) [inline src/bin/sota_provisioner.rs]: unstable api?

File: src/bin/sota_provisioner.rs [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+            file.write_all(&root_meta_bytes).map_err(|_| ())
+        } else {
+            Ok(())
+        }
+    }
+
+    /// Checks the local FS for the `root.json`, returns `true` if it is present, `false`
+    /// if it is not, and `Err` if the check fails entirely.
+    fn check_fs_for_root_json(&self) -> Result<bool, ()> {
+        Ok(false) // TODO
```
Comment (taheris) [inline src/bin/sota_provisioner.rs]: do stuff

Comment (jochenschneider) [inline src/bin/sota_provisioner.rs]: https://doc.rust-lang.org/std/path/struct.Path.html#method.exists

File: tests/toml/old.toml [hunk 3]
```diff
@@ -43,3 +43,5 @@ websocket_server = "127.0.0.1:3012"
 client = "http://127.0.0.1:8901"
 storage_dir = "/var/sota"
 timeout = 20
+
+[uptane]
```
Comment (cajun-rat) [inline tests/toml/old.toml]: Isn't this config here to check that sota.toml files are backwards compatible?

**Comment by taheris** [comment]
Superseded by PR #169

**Pull request closed without merging**